src/Type/Laminas/PluginMethodDynamicReturnTypeExtension/AbstractPluginMethodDynamicReturnTypeExtension.php

```diff
@@ -36,18 +39,25 @@ abstract class AbstractPluginMethodDynamicReturnTypeExtension implements Dynamic
         MethodCall $methodCall,
         Scope $scope
     ): Type {
-        $pluginManager = $this->serviceManagerLoader->getServiceLocator($this->getPluginManagerName());
-        $argType       = $scope->getType($methodCall->args[0]->value);
-        if (! $argType instanceof ConstantStringType) {
-            throw new \PHPStan\ShouldNotHappenException(\sprintf('Argument passed to %s::%s should be a string, %s given',
-                $methodReflection->getDeclaringClass()->getName(),
-                $methodReflection->getName(),
-                $argType->describe(VerbosityLevel::typeOnly())
-            ));
+        $argType = $scope->getType($methodCall->args[0]->value);
+        $strings = TypeUtils::getConstantStrings($argType);
```

Comment:
@neilime why not simply`$argType instanceof ConstantStringType` ?

Reply:
This is actually better. It accounts for `'foo'|'bar'` etc.